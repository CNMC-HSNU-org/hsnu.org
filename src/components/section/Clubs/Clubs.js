import React from "react"
import Swiper from "react-id-swiper"
import { Container, Row, Col } from "react-bootstrap"

import spotify from "../../../images/icons/spotify-colored.svg"
import facebook from "../../../images/icons/facebook-colored.svg"
import telegram from "../../../images/icons/telegram-colored.svg"
import instagram from "../../../images/icons/instagram-colored.svg"
import patreon from "../../../images/icons/patreon-colored.svg"
import soundcloud from "../../../images/icons/soundcloud-colored.svg"
import twitter from "../../../images/icons/twitter-colored.svg"
import youtube from "../../../images/icons/youtube-colored.svg"

import asylum from "../../../images/cover/asylum.jpeg"

// style
import "./Clubs.scss"

const Clubs = () => {
  // config of swiper
  const params = {
    loop: "true",
    slidesPerView: "auto",
    navigation: {
      nextEl: ".swiper-button-next",
    },
  }

  return (
    <section id="clubs">
      <h2 className={"is-2 serif bold"}>社團</h2>
      <Swiper {...params}>
        {[1, 2, 3].map(item => (
          <div>
            <div className={"club-card"}>
              {/* cover */}
              <figure>
                <img src={asylum} />
              </figure>

              {/* club info card */}
              <div className={"club-info"}>
                {/* title */}
                <h3 className={"is-3 bold serif"}>英語會話社</h3>

                {/* social media */}
                <Container>
                  <Row className={"icons"}>
                    {[
                      telegram,
                      spotify,
                      facebook,
                      instagram,
                      soundcloud,
                      youtube,
                      patreon,
                      twitter,
                    ].map(icon => (
                      <Col className={"icon col-4 col-sm-3"}>
                        <a hre="#">
                          <img src={icon} />
                        </a>
                      </Col>
                    ))}
                  </Row>
                </Container>
              </div>
            </div>
          </div>
        ))}
      </Swiper>
    </section>
  )
}

export default Clubs
