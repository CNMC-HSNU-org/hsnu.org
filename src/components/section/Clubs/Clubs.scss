#clubs {
  // box
  width: 80%;
  height: auto;
  margin: 2.5rem auto;

  // position
  position: relative;

  // title
  h2 {
    margin: 2rem 0;
  }

  // 1st layer
  .swiper-container {
    // box
    width: 100%;
    height: 100%;
  }

  // Club Card Large (Component)
  .swiper-slide {
    // box
    width: 100%;
    height: 100%;

    // flex align
    display: flex;
    flex-direction: column;
    align-items: center;

    .club-card {
      // box
      width: 90%;
      max-height: 100%;
      margin: 1.5rem 0;

      // cover
      figure {
        width: 100%;
        height: auto;
        max-height: 50%;

        // image itself
        img {
          width: 100%;
          height: 100%;
          object-fit: cover;
        }
      }

      // club info
      .club-info {
        // box
        width: 100%;
        height: auto;
        padding: 2rem;
        margin: 1rem 0;
        background: #ffffff;
        border: solid 1px #3c3c3c;

        // name
        h3 {
          margin-bottom: 1.5rem;
          text-align: center;
        }

        // icons
        .icons {
          .icon {
            height: 2rem;
            width: 2rem;
            padding: 0;
            margin: 1.5rem 0;

            img {
              height: 100%;
              width: 100%;
            }
          }
        }
      }
    }
  }

  // navigation button
  .swiper-button-next {
    background-image: url("../../../images/icons/functional/next-min.svg");

    // position
    top: 70%;
    transform: translateY(-50%);
    z-index: 1;
  }
}

@media (min-width: 992px) {
  #clubs {
    // box
    height: auto;
    width: 90%;
    margin: 4rem auto;

    .swiper-wrapper {
      left: -10%;
    }

    .swiper-slide {
      width: 80%;

      .club-card {
        position: relative;

        // cover
        figure {
          height: 60vh;
          width: 90%;
          max-height: 100%;
        }

        .club-info {
          // box
          width: 50%;
          border: none;
          box-shadow: 4px 5px 20px rgba(0, 0, 0, 0.3);

          // position
          position: absolute;
          right: 0;
          top: 50%;
          transform: translateY(-50%);
        }
      }
    }

    // navigation button
    .swiper-button-next {
      // position
      top: 50%;
      right: 30%;
      transform: translate(-50%, -50%);
    }
  }
}
