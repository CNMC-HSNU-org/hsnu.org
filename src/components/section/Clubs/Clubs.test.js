import React from "react"
import renderer from "react-test-renderer"
import { ClubsPure } from "./Clubs"

describe("Section/Clubs", () => {
  it("renders correctly", () => {
    const clubs = [
      {
        node: {
          title: "新聞社",
          acf: {
            image: {
              sizes: {
                large:
                  "https://wordpress.hsnu.org/wp-content/uploads/2020/03/Screenshot_2020-03-19-關於我們-報導者-The-Reporter-1024x495.png",
              },
            },
            social_medias: [
              {
                media_type: "facebook",
                media_link: "https://www.facebook.com/twreporter/",
              },
              {
                media_type: "line",
                media_link: "https://line.me/ti/p/%40nbs5015j",
              },
              {
                media_type: "patreon",
                media_link: "https://www.instagram.com/twreporter/",
              },
            ],
          },
        },
      },
    ]
    const tree = renderer.create(<ClubsPure clubs={clubs} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
