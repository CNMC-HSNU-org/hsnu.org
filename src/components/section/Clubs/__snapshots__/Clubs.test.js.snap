// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section/Clubs renders correctly 1`] = `
<section
  id="clubs"
>
  <h2
    className="is-2 bold"
  >
    社團
  </h2>
  <span
    className=""
    style={
      Object {
        "display": "inline-block",
        "height": 0,
        "width": 0,
      }
    }
  />
</section>
`;
